packages/xchain-thorchain/src/util.ts

```diff
@@ -14,5 +18,86 @@ export const getDenom = (v: Asset): string => {
  */
 export const getAsset = (v: string): Asset | null => {
   if (v === getDenom(AssetRune)) return AssetRune
-  return null
+  return assetFromString(`${THORChain}.${v.toUpperCase()}`)
+}
+
+/**
+ * Type guard for transfer event
+ */
+export const isTransferEvent = (v: TxEvent): boolean => v.type === 'transfer'
+
+/**
+ * Type guard for recipient attribute
+ */
+export const isRecipient = (v: TxEventAttribute): boolean => v.key === 'recipient'
+
+/**
+ * Type guard for sender attribute
+ */
+export const isSender = (v: TxEventAttribute): boolean => v.key === 'sender'
+
+/**
+ * Type guard for amount attribute
+ */
+export const isAmount = (v: TxEventAttribute): boolean => v.key === 'amount'
+
+/**
+ * Parse amount string to value and denom
+ */
+export const parseAmountString = (v: string): TransferAmount | undefined => {
+  try {
+    const value = parseInt(v)
+    const denom = v.replace(value.toString(), '')
+    return { value, denom }
+  } catch (e) {
+    return undefined
+  }
+}
```

Comment:
Using `parseInt` have some limitations in JS. We can avoid it by parsing `value`, `denom` using `RegEx`. To store values more type safety we can also use updated type `TransferAmount` (see previous comment).